### ISSUE TITLE
SSR support?

### ISSUE THREAD

**Issue by march08** [mhart/aws4fetch#22]
works well in the browser however there are some differences in the WebAPI vs node. Are you planning to add SSR support as well? Thanks

**Comment by mhart** [comment]
@march08 I'm not sure what you mean? Server Side Rendering is orthogonal to a library like this. Can you be more specific?

**Comment by march08** [comment]
> @march08 I'm not sure what you mean? Server Side Rendering is orthogonal to a library like this. Can you be more specific?

For example, `crypto.subtle` doesn't exist on server (node) but web crypto api only. 
`headers.keys` is undefined on server as well.

**Comment by mhart** [comment]
Right, Node.js doesn't support `fetch` (you could try out https://github.com/node-fetch/node-fetch ) or `crypto.subtle` (you could try out https://github.com/PeculiarVentures/node-webcrypto-ossl)

**Comment by march08** [comment]
fetch can be added with `isomorphic-fetch` but `Headers.keys()` doesn't work. In the end I had to rewrite the code to make it work )

**Comment by mhart** [comment]
Well that must be a bug with that library, it's supported in node-fetch:

https://github.com/node-fetch/node-fetch/blob/ca50c22e3410ee288a7b4a8a89384d6df95ae3b6/%40types/index.d.ts#L46

**Comment by rafaelnogueira1** [comment]
@march08 how did you solve this? 'm using nextjs and i stuck in this problem.

**Comment by rafaelnogueira1** [comment]
In the end i end up change to use aws4-axios to solve my problem.

**Comment by badalya1** [comment]
@march08  You are probably looking for [aws4](https://github.com/mhart/aws4) if you want to use it in node.

**Comment by half2me** [comment]
@badalya1 well I want to use it on browser and cf workers in production, but in node just for testing.

**Comment by mhart** [comment]
Changing this to be a Node.js-support specific issue. The ability to use this locally should be supported by [`workerd`](https://github.com/cloudflare/workerd) and I assume Deno as well (though haven't checked).

If anyone knows what the specific issues with current Node.js support of these APIs are, let me know!

**Comment by n1ru4l** [comment]
I achieved Node.js support with `@whatwg-node/fetch`, we run it on both Node.js and Cloudflare Workers.

```patch
diff --git a/dist/aws4fetch.esm.js b/dist/aws4fetch.esm.js
index 9a47c421432cb8dc9f92a07a49fd46d2c00c57f3..f853ca70741083272fea7e57b405c9920402be3e 100644
--- a/dist/aws4fetch.esm.js
+++ b/dist/aws4fetch.esm.js
@@ -1,3 +1,5 @@
+import { fetch, Request, Headers, crypto, TextEncoder } from '@whatwg-node/fetch';
+
 /**
  * @license MIT <https://opensource.org/licenses/MIT>
  * @copyright Michael Hart 2022
diff --git a/dist/aws4fetch.esm.mjs b/dist/aws4fetch.esm.mjs
index 9a47c421432cb8dc9f92a07a49fd46d2c00c57f3..f853ca70741083272fea7e57b405c9920402be3e 100644
--- a/dist/aws4fetch.esm.mjs
+++ b/dist/aws4fetch.esm.mjs
@@ -1,3 +1,5 @@
+import { fetch, Request, Headers, crypto, TextEncoder } from '@whatwg-node/fetch';
+
 /**
  * @license MIT <https://opensource.org/licenses/MIT>
  * @copyright Michael Hart 2022
  * 
```

**Comment by 0x1abin** [comment]
It works on both node v20.5.1 and Cloudflare Worker.
